lib/virt_autotest/utils.pm

```diff
@@ -439,31 +439,22 @@ sub wait_guest_online {
     }
 }
 
-# Shutdown all guests and wait until they are shutdown
+# Shutdown all guests
 sub shutdown_guests {
-    ## Reboot the guest to ensure the settings are applied
-    # Shutdown and start the guest because some might have the on_reboot=destroy policy still applied
-    script_run("virsh shutdown $_") foreach (keys %virt_autotest::common::guests);
-    # Wait until guests are terminated
-    wait_guests_shutdown();
-}
-
-# wait_guests_shutdown([$timeout]) waits for all guests to be shutdown
-sub wait_guests_shutdown {
-    my $retries = shift // 240;
-    # Note: Domain-0 is for xen only, but it does not hurt to exclude this also in kvm runs.
-    # Firstly wait for guest shutdown for a while, turn it off forcibly using "virsh destroy" if timed-out.
-    # Then wait for guest shutdown again with default "die => 1".
-    if (script_retry("! virsh list | grep -v Domain-0 | grep running", timeout => 60, delay => 1, retry => $retries, die => 0) != 0) {
-        script_run("virsh destroy $_") foreach (keys %virt_autotest::common::guests);
+    my @guests = @_;
+    foreach my $guest (@guests) {
+        script_run("virsh shutdown $guest");
+        script_retry("virsh domstate $guest|grep 'shut off'", retry => 5);    #assert if vm fails shutdown
```

Comment:
Yes, I checked it too last week and saw the same behavior. It might be a bug in script_retry.  I could not find the root cause. I will not change the code here.